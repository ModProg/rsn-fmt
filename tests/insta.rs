use include_dir::include_dir;

const INPUTS: include_dir::Dir = include_dir!("$CARGO_MANIFEST_DIR/tests/input");

#[test]
fn format() {
    for file in INPUTS.entries() {
        let contents = file.as_file().unwrap().contents_utf8().unwrap();
        // TODO let before: OwnedValue = rsn::from_str(contents).unwrap();

        let formatted = rsn_fmt::format_str(&contents);

        println!("{formatted}");
        // let after: OwnedValue = rsn::from_str(formatted).unwrap();
        // assert_eq!(before, after);

        insta::assert_snapshot!(file.path().display().to_string(), formatted);
    }
}
