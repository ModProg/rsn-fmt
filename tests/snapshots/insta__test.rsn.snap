---
source: tests/insta.rs
expression: formatted
---
Hello {
    hello: hi,

    PI: [3.14, 3],
}

